coordinator/src/trade/mod.rs

```diff
@@ -173,6 +207,66 @@ impl TradeExecutor {
         };
     }
 
+    /// Settles the accepted invoice for the given trader
+    async fn settle_invoice(&self, trader: PublicKey, order_id: Uuid) -> Result<()> {
+        let pre_image = spawn_blocking({
+            let pool = self.node.pool.clone();
+            move || {
+                let mut conn = pool.get()?;
+                let pre_image =
+                    db::hodl_invoice::update_hodl_invoice_to_settled(&mut conn, order_id)?;
+
+                anyhow::Ok(pre_image)
+            }
+        })
+        .await??
+        .context("Missing pre_image")?;
+
+        let pre_image = PreImage::from_url_safe_encoded_pre_image(&pre_image)?;
+
+        self.node
+            .lnd_bridge
+            .settle_invoice(pre_image.get_base64_encoded_pre_image())
+            .await?;
+
+        tracing::info!(%trader, hash = pre_image.hash, "Settled invoice");
+
+        Ok(())
+    }
+
+    /// Cancels a potential pending offer if the proposal failed.
+    async fn cancel_offer(&self, trader: PublicKey) -> Result<()> {
+        if let Some(channel) = self
+            .node
+            .inner
+            .get_dlc_channel(|channel| channel.get_counter_party_id() == to_secp_pk_29(trader))?
+        {
+            self.node.process_dlc_message(
+                trader,
+                &TenTenOneMessage::Reject(TenTenOneReject {
+                    reject: Reject {
+                        channel_id: channel.get_id(),
+                        timestamp: OffsetDateTime::now_utc().unix_timestamp() as u64,
+                        reference_id: None,
+                    },
+                }),
+            )?;
+
+            spawn_blocking({
+                let pool = self.node.pool.clone();
+                move || {
+                    let mut conn = pool.get()?;
+                    db::last_outbound_dlc_message::delete(&mut conn, &trader)?;
+
+                    anyhow::Ok(())
+                }
+            })
+            .await??;
```

Comment:
🔧 This is a bit scary. Imagine failing to delete the message or deleting the wrong message.

Could we instead insert the channel-opening offer only _after_ we settle the invoice?

Reply:
I don't think that's easily possible.. you are right that we could end up deleting an unrelated last outbound dlc message. The consequences though should be insignificant, since this is only used for recovery reasons.

Reply:
Also note, we are only doing this when the dlc channel is offered. So any `last_outbound_dlc_message` would anyways be useless.